### PR TITLE
Feature/create liquidity pos and uniswap pool improvements

### PR DESCRIPTION
**Pull request by unibeck**
This PR includes several additions and improvements:
- Added BuffChain account and updated hardhat and tests to use its private key
- New deployment script to provide accounts with their due TUFF amount (and adjusted tests accordingly)
- Updated uniswap pool deployment to include creating a liquidity position
  - This was needed for go live
  - Liquidity is provided by BuffChain's cut
  - This enables more comprehensive testing which requires TUFF swaps which was possible before due to the lack of liquidity
- Added test for uniswap pool and for liquidity position. While we do not have a contract for these, they certainly need to be tested as part of our deployment pipeline
- Updated TuffToken contract to be consistent with OpenZeppelin ERC20 template. This was to fix a STF bug found while implementing uniswap liquidity/swaps
  - Added dev docs in accordance to OpenZeppelin
  - More needs to be done here to make sure our token is standard (outside the Diamond and fee taking parts)
- Enhanced Typescript support
  - migrated deployments over to typescript
  - refined typechain (type support for contracts)
- Updated github action to use npm instead of yarn

### CODE REVIEW

File: test/unit_tests/tokenMaturity.js [hunk 1]
```diff
@@ -178,14 +161,15 @@ describe('TokenMaturity', function () {
         expect(senderTuffTokenBalanceAfterRedemption).to.equal(0,
             "Holder's balance was not reset");
 
+        // await tuffTokenDiamond.redeem();
```
Comment (ianballard) [inline test/unit_tests/tokenMaturity.js:164]: reason for commenting this out?

File: utils/consts.js [hunk 2]
```diff
@@ -29,17 +26,15 @@ let constsMap = {
         //Uniswap (https://docs.uniswap.org/protocol/reference/deployments)
         UNISWAP_V3_ROUTER_ADDR: "0xE592427A0AEce92De3Edee1F18E0157C05861564",
         UNISWAP_V3_FACTORY_ADDR: "0x1F98431c8aD98523631AE4a59f267346ea31F984",
+        UNISWAP_V3_NonfungiblePositionManager_ADDR: "",
```
Comment (ianballard) [inline utils/consts.js:29]: can we keep the naming consistent with all caps and underscores?

**Comment by ianballard** [comment]
looks great! awesome work. couple small questions.

File: test/unit_tests/tokenMaturity.js (hunk 1 above)
Comment (unibeck) [reply test/unit_tests/tokenMaturity.js:164]: I was debugging the expect revert call below. You can see this change was an addition, but it never should have been. I'll remove it

File: utils/consts.js (hunk 2 above)
Comment (unibeck) [reply utils/consts.js:29]: You got it!

**Review by ianballard — APPROVED** [review]

**Pull request merged**